### PR TITLE
(test) O3-2514 : Setup the template repository with playwright

### PR DESCRIPTION
**Pull request by RandilaP**
## Requirements
- [ ] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.

## Summary
The goal of this PR is to set up the esm template repository for E2E testing with playwright

## Related Issue

https://issues.openmrs.org/browse/O3-2514

### CODE REVIEW

**Comment by RandilaP** [comment]
@jayasanka-sack Could you please review this

File: e2e/specs/initiate-test.spec.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import test from "@playwright/test";
+import { HomePage } from "../pages";
+import { expect } from "@playwright/test";
+
+test("initiate test", async ({ page}) => {
```
Comment (jayasanka-sack) [inline e2e/specs/initiate-test.spec.ts:5]:
```suggestion
test("Sample test", async ({ page}) => {
```

File: e2e/specs/initiate-test.spec.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    const homePage = new HomePage(page);
+    await homePage.goto();
+    await expect(homePage.page.getByRole('link', { name: 'Home' })).toBeVisible();
+});
```
Comment (jayasanka-sack) [inline e2e/specs/initiate-test.spec.ts:9]: Add a newline at the end of the file

File: e2e/README.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+## Debugging Tests
+
+Refer to [this documentation](https://playwright.dev/docs/debug) on how to debug a test.
+
+## Configuration
+
+This is very much underdeveloped/WIP. At the moment, there exists a (git-shared) `.env`
+file which can be used for configuring certain test attributes. This is most likely
+about to change in the future. Stay tuned for updates!
```
Comment (jayasanka-sack) [inline e2e/README.md:93]: Add the `.env.example` file to the root. Also update the root readme file with instructions to copy the `.env.example` file and create `.env` file by running `cp .env.example .env`). Link to this readme file too.

File: e2e/README.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+and explore the details of individual tests, including any errors or failures, video 
+recordings, and the steps involved in each test. Simply click on a test to view its details.
+
+## Debugging Tests
+
+Refer to [this documentation](https://playwright.dev/docs/debug) on how to debug a test.
+
+## Configuration
+
+This is very much underdeveloped/WIP. At the moment, there exists a (git-shared) `.env`
```
Comment (jayasanka-sack) [inline e2e/README.md:91]:
```suggestion
This is very much underdeveloped/WIP. At the moment, there exists a (git-shared) `.env.example`
```

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]
Thanks @RandilaP. I've added some comments. Let me know what you think.

**Comment by jayasanka-sack** [comment]
Also, update the jest config file to skip playwright test files (that's why the build is failing). 

`testPathIgnorePatterns: [path.resolve(__dirname, 'e2e')],`
`collectCoverageFrom....`

**Comment by RandilaP** [comment]
@jayasanka-sack Could you please review this

**Comment by jayasanka-sack** [comment]
@RandilaP the build is still failing. could you please check?

**Comment by RandilaP** [comment]
@jayasanka-sack Could you please approve the workflow

**Review by jayasanka-sack — APPROVED** [review]
Thanks @RandilaP !

**Pull request merged**